Sources/XCTest/XCTestCase.swift

```diff
@@ -92,6 +99,8 @@ extension XCTestCase {
 
                 XCTPrint("Test Case '\(fullName)' started.")
 
+                observationCenter.testCaseWillStart(testCase)
```

Comment:
As it turns out, Apple XCTest does exactly that (well, not *exactly*, because the logging uses the legacy `XCTestObserver` protocol, but close enough. I learned all this while working through https://github.com/Quick/Quick/pull/486)

I would like to make that refactor, yes. In fact, one reason why I wrote the observer in the functional test to keep track of state in properties instead of just `print`ing was to facilitate this, since `XCTestObservationCenter` doesn't provide any guarantees of the order in which observers are called.